src/Strategy/Cocoapods/PodfileLock.hs

```diff
@@ -23,89 +26,149 @@ import Data.HashMap.Lazy qualified as HashMap (toList)
 import Data.Map.Strict (Map)
 import Data.Map.Strict qualified as Map
 import Data.Set (Set)
+import Data.String.Conversion (toString, toText)
 import Data.Text (Text)
 import Data.Void (Void)
 import Data.Yaml ((.:), (.:?))
 import Data.Yaml qualified as Yaml
 import DepTypes (DepType (GitType, PodType), Dependency (..), VerConstraint (CEq))
+import Effect.Exec (AllowErr (..), Command (..), Exec, exec, execJson)
 import Effect.Grapher (LabeledGrapher, direct, edge, label, withLabeling)
 import Effect.ReadFS (ReadFS, readContentsYaml)
-import Graphing (Graphing)
+import Graphing (Graphing, gtraverse)
 import Options.Applicative (Alternative ((<|>)))
-import Path
-import Text.Megaparsec (MonadParsec (takeWhileP), Parsec, between, empty, errorBundlePretty, parse, some, takeWhile1P)
+import Path (Abs, Dir, File, Path, parent)
+import System.FilePath ((<.>), (</>))
+import Text.Megaparsec (Parsec, between, empty, errorBundlePretty, parse, some, takeWhile1P, takeWhileP)
 import Text.Megaparsec.Char (char)
 import Text.Megaparsec.Char.Lexer qualified as L
+import Data.Text.Extra (showT)
 
-analyze' :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs File -> m (Graphing Dependency)
+analyze' :: (Has ReadFS sig m, Has Diagnostics sig m, Has Exec sig m) => Path Abs File -> m (Graphing Dependency)
 analyze' file = do
   podfileLock <- readContentsYaml file
-  context "Building dependency graph" $ pure (buildGraph podfileLock)
+  context "Building dependency graph" $ buildGraph file podfileLock
```

Comment:
I know this is still wip, but this caught my eye.

Prefer the use of alternatives, so it's easier to reason on limitations, and warnings can be embedded outside of graphing logic.
 
```haskell
-- Function and variable names, etc. for example. 
-- Semmatically there will likely be better names than what I've 

podLockDynamicAnalysis podLockFile = do
 someGraph <- someBuildGraph podLockFile
 graph <- warnOnErr MissingVendorReplacements (enrichGraph someGraph)  <||> pure someGraph
 pure graph

data PodlockGraphEnrichmentFailed = PodlockGraphEnrichmentFailed (Path Abs File)

instance ToDiagnostic MissingVendorReplacements where
  renderDiagnostic (MissingVendorReplacements podLockFile) = "..."
```